stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -1415,3 +1415,32 @@ function normalize(v::AbstractVector, p::Real = 2)
         return T[]
     end
 end
+
+"""
+    eachrow(A::AbstractVecOrMat)
+
+Create a generator iterating over views of A's first dimension.
+See also [`eachcol`](@ref) and [`eachslice`](@ref).
+"""
+eachrow(A::AbstractVecOrMat) = (view(A, i, :) for i in axes(A, 1))
+
+
+"""
+    eachcol(A::AbstractVecOrMat)
+
+Create a generator iterating over views of A's second dimension.
+See also [`eachrow`](@ref) and [`eachslice`](@ref).
+"""
+eachcol(A::AbstractVecOrMat) = (view(A, :, i) for i in axes(A, 2))
+
+"""
+    eachslice(A; dims = 1)
+
+Create a generator iterating over views of A's nth dimension.
+See also [`eachrow`](@ref) and [`eachcol`](@ref).
+"""
+@inline function eachslice(A; dims = 1)
```

Comment:
Maybe `dims::Integer`?

Reply:
Not sure I see the value. A non-integer `dims` would fail anyway.

Reply:
It's also possible to eventually extend this implementation to iterating over many dimensions.  For example, `eachslice(reshape(1:24, 4, 3, 2); dims = (2, 3))` could meaningfully generalize to iterate over both the second and third dimension and thus return the columns.  Let's not worry about that for now — but let's give ourselves the space to develop that new capability by ensuring that it's an error to pass more than one dimension.  And let's support passing a tuple of one dimension — the keyword name is dim**s**, after all:

```julia
length(dims) == 1 || throw(ArgumentError("only single dimensions are supported"))
dim = first(dims)
# … remainder uses `dim` not `dims`
```

Reply:
Done.